src/Scriban/Parsing/Lexer.cs

```diff
@@ -1037,7 +1068,14 @@ private bool ReadCode()
                     }
 
                     bool specialIdentifier = c == '$';
-                    if (IsFirstIdentifierLetter(c) || specialIdentifier)
+                    char nextChar = PeekChar(1); //Used for checking if the dollar is before the string
+                    if (specialIdentifier &&  (nextChar == '"' || nextChar == '\''))
+                    {
+                        NextChar(); //Skip '$'
```

Comment:
We probably don't want to skip `$` here because we want to capture the `start` position correctly from `ReadInterpolatedString` below